### PR TITLE
9 task device handler

### PR DESCRIPTION
**Pull request by ThatsTheEnd**
## Description

Adds an initial draft to the device handler that can start the ICL but not stop it atm

## Related Issue

#9 

## Type of Change

- [ ] 📚 Examples / docs / tutorials / dependencies update
- [ ] 🔧 Bug fix (non-breaking change which fixes an issue)
- [ ] 🥂 Improvement (non-breaking change which improves an existing feature)
- [X] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔐 Security fix

## Checklist

- [ ] I've read the [`CODE_OF_CONDUCT.md`](https://github.com/ThatsTheEnd/horiba-python-sdk/blob/master/CODE_OF_CONDUCT.md) document.
- [ ] I've read the [`CONTRIBUTING.md`](https://github.com/ThatsTheEnd/horiba-python-sdk/blob/master/CONTRIBUTING.md) guide.
- [x] I've updated the code style using `make codestyle`.
- [x] I've written tests for all new methods and classes that I created.
- [x] I've written the docstring in Google format for all the methods and classes that I used.

### CODE REVIEW

File: horiba_sdk/communication/abstracts.py
Comment (w-samuelgauthier) [inline horiba_sdk/communication/abstracts.py:1]: Should we rename this file to `abstract_communicator.py`?

File: horiba_sdk/devices/device_manager.py
Comment (w-samuelgauthier) [inline horiba_sdk/devices/device_manager.py:1]: I'm not sure if I understood everything correctly and if it's not overkill what I think. Couldn't we have:
- a class `ACL` to wrap the ACL communication (with `start` and `stop`)
- a class `Devices` or `DeviceList` that contains the devices, pass it in the constructor an `AbstractCommunicator` and has the `discover_devices` function
What do you think?

**Review by w-samuelgauthier — APPROVED** [review]
Looks good!

File: horiba_sdk/communication/abstracts.py
Comment (ThatsTheEnd) [reply horiba_sdk/communication/abstracts.py:1]: Good catch, I originally had one more abstract class in it. But atm it makes sense :)
Will change it in the next commit.

**Pull request merged**